src/sidebar/components/hooks/unsaved-changes.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect } from 'preact/hooks';
```

Comment:
Next week I will look at upstreaming this into the frontend-shared library.

Reply:
Actually, we already have [a similar hook](https://github.com/hypothesis/frontend-shared/blob/4cd957e560fb515246a95d20f7b75a02b9c80828/src/hooks/use-warn-on-page-unload.ts) in frontend-shared that we currently use in LMS (IIRC).

However, this hook here has a nice addition that prevents the listener to be configured more than once, so we should probably just replace frontend-shared's hook with this one (or at least its implementation).

The frontend-shared one also has the issue of setting `returnValue` to an empty string.

Reply:
> However, this hook here has a nice addition that prevents the listener to be configured more than once, so we should probably just replace frontend-shared's hook with this one (or at least its implementation).

Yes. This means we now have three implementations: this one, one in h and frontend-shared. I'll look to consolidate them all later today.